osu.Game.Rulesets.Osu/Objects/Drawables/SliderInputManager.cs

```diff
@@ -0,0 +1,254 @@
+        /// </summary>
+        private readonly List<OsuAction> lastPressedActions = new List<OsuAction>();
+
+        private Vector2? screenSpaceMousePosition;
+        private readonly DrawableSlider slider;
+
+        public SliderInputManager(DrawableSlider slider)
+        {
+            this.slider = slider;
+        }
+
+        /// <summary>
+        /// This component handles all input of the slider, so it should receive input no matter the position.
+        /// </summary>
+        public override bool ReceivePositionalInputAt(Vector2 screenSpacePos) => true;
+
+        protected override bool OnMouseMove(MouseMoveEvent e)
+        {
+            screenSpaceMousePosition = e.ScreenSpaceMousePosition;
+            return base.OnMouseMove(e);
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+            updateTracking(isMouseInFollowArea(Tracking));
+        }
+
+        public void PostProcessHeadJudgement(DrawableSliderHead head)
+        {
+            if (!head.Judged || !head.Result.IsHit)
+                return;
+
+            if (!isMouseInFollowArea(true))
+                return;
+
+            Debug.Assert(screenSpaceMousePosition != null);
+
+            Vector2 mousePositionInSlider = slider.ToLocalSpace(screenSpaceMousePosition.Value) - slider.OriginPosition;
+
+            // When the head is hit and the mouse is in the expanded follow area, force a hit on every nested hitobject
+            // from the start of the slider that is within the follow area.
+
+            bool forceMiss = false;
+
+            foreach (var nested in slider.NestedHitObjects.OfType<DrawableOsuHitObject>())
+            {
+                // Skip nested objects that are already judged.
+                if (nested.Judged)
+                    continue;
+
+                // Stop the process when a nested object is reached that can't be hit before the current time.
+                if (nested.HitObject.StartTime > Time.Current)
+                    break;
+
+                float radius = getFollowRadius(true);
+                double objectProgress = Math.Clamp((nested.HitObject.StartTime - slider.HitObject.StartTime) / slider.HitObject.Duration, 0, 1);
+                Vector2 objectPosition = slider.HitObject.CurvePositionAt(objectProgress);
+
+                // When the first nested object that is further outside the follow area is reached,
+                // forcefully miss all other nested objects that would otherwise be valid to be hit.
+                // This covers a case of a slider overlapping itself that requires tracking to a tick on an outer edge.
+                if (forceMiss || (objectPosition - mousePositionInSlider).LengthSquared > radius * radius)
+                {
+                    nested.MissForcefully();
+                    forceMiss = true;
+                }
+                else
+                    nested.HitForcefully();
+            }
```

Comment:
So, I asked what was preferred and it seems the suggested change here would be something like this:
```C#
            // Check if all past ticks are in range
            foreach (var nested in slider.NestedHitObjects.OfType<DrawableOsuHitObject>())
            {
                // Skip nested objects that are already judged.
                if (nested.Judged)
                    continue;

                // Stop the process when a nested object is reached that can't be hit before the current time.
                if (nested.HitObject.StartTime > Time.Current)
                    break;

                float radius = getFollowRadius(true);
                double objectProgress = Math.Clamp((nested.HitObject.StartTime - slider.HitObject.StartTime) / slider.HitObject.Duration, 0, 1);
                Vector2 objectPosition = slider.HitObject.CurvePositionAt(objectProgress);

                // If any nested object that is further outside the follow area is reached,
                // forcefully miss all nested objects.
                if ((objectPosition - mousePositionInSlider).LengthSquared > radius * radius)
                {
                    forceMiss = true;
                    break;
                }
            }

            // Validate or miss all past ticks
            foreach (var nested in slider.NestedHitObjects.OfType<DrawableOsuHitObject>())
            {
                if (nested.Judged)
                    continue;

                if (nested.HitObject.StartTime > Time.Current)
                    break;

                if (forceMiss)
                    nested.MissForcefully();
                else
                    nested.HitForcefully();
            }
```

Reply:
Thanks for the feedback, but strongly disagree with this one, we'll go with what we have.

Reply:
I don't think I mind much either way, but this is an extreme edge case imo.

Reply:
Sorry, the reaoning for that was here : https://github.com/ppy/osu/pull/25776#discussion_r1428966350
Basically, it doesn't make sense for the 2nd example above (Late hit outside tracking radius https://drive.google.com/file/d/10gdU9WV2ze0jRVaDiqov1I-jpgJMO3q1/view?usp=share_link) to be full misses but the 4th example above (Sliders with outside extremums https://drive.google.com/file/d/10hvIwpzszqceWP-5NI6-WMcFoJDZJ-gc/view) to be hitting early ticks.
Because in both cases, the ball has left the follow-circle range at one point.

Reply:
I applied this change in the recent commits.